enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,338 @@
+Using the client is not an option because that requires a local Prometheus
+instance, not viable in the typical resource constrained MicroShift deployment.
+The direct write requests take raw Prometheus data and require both special
+labels (`_id` for the cluster id) and crafting the authentication headers
+(cluster id and pull secret in a specific format in HTTP headers).
+
+A comprehensive list of metrics sent by MicroShift follows.
+
+#### `cluster:capacity_cpu_cores:sum`
+This metric reports the CPU cores MicroShift can use. Already supported in
+OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `label_kubernetes_io_arch`: Label from the node, determines the architecture.
+  Values can be `amd64`, `aarch64`.
+* `label_node_openshift_io_os_id`: Label from the node, determines the OS type.
+  Values can be `rhel`.
+* `label_beta_kubernetes_io_instance_type`: Label from the node. MicroShift
+  sets this to `rhde`. Possible values are `rhde`.
+
+#### `cluster:capacity_memory_bytes:sum`
+This metric reports the memory MicroShift can use in bytes. Already supported
+in OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `label_kubernetes_io_arch`: Label from the node, determines the architecture.
+  Values can be `amd64`, `aarch64`.
+* `label_node_openshift_io_os_id`: Label from the node, determines the OS type.
+  Values can be `rhel`.
+* `label_beta_kubernetes_io_instance_type`: Label from the node. MicroShift
+  sets this to `rhde`. Possible values are `rhde`.
+
+#### `cluster:cpu_usage_cores:sum`
+This metric reports the CPU usage in percentage. Already supported in
+OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+
+#### `cluster:memory_usage_bytes:sum`
+This metric reports the memory usage in percentage. Already supported in
+OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+
+#### `cluster:usage:resources:sum`
+This metric reports the number of Kubernetes objects by resource type. Already
+supported in OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `resource`: name of the kubernetes resource. Values can be `pods`,
+  `namespaces`, `services`, `ingresses.networking.k8s.io`,
+  `routes.route.openshift.io`,
+  `customresourcedefinitions.apiextensions.k8s.io`.
+
+#### `cluster:usage:containers:sum`
+This metric reports the number of running containers. Already supported in
+OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+
+#### `microshift_version`
+This metric reports information about MicroShift version. New metric not
+supported in OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `version`: MicroShift version. Possible values are `\d{1}.\d{1,2}.\d{1,2}`.
+* `deployment_type`: Describes deployment type. Possible values are `rpm`,
+  `ostree`, `bootc`.
+* `deployment_id`: In case `deployment_type` is `ostree`, this will hold the
+  ostree commit. In case of `bootc`, include image SHA. Format is a 64
+  character SHA.
```

Comment:
what's the cardinality of this label? E.g. given 10,000 MicroShift instances reporting to Telemetry, how many different values can we expect?

Reply:
> Each customer will have their own id's because they customize base images/commits. So a rough estimation for the different values for this would be number of customers * number of different microshift versions they have.

This is too much cardinality for Telemetry I'm afraid. I'm also not sure how the value will be useful for reporting? If you're interested in knowing how many clusters run a "vanilla" version vs. how many run a customized version then I would encourage to encode this information in the metric (e.g. . `microshift_version{flavor="custom|stock",../}`)Not sure if it's feasible though..

Reply:
After some further discussion, ended up removing this label and will investigate alternatives for future releases.